qa-tester/assets/src/js/admin-bar/components/container/index.js

```diff
@@ -0,0 +1,241 @@
+/**
+ * WordPress dependencies
+ */
+import React, { Component } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import './index.css';
+import {
+	getProtectedBranches,
+	getPullRequestsWithBuilds,
+} from '../../utils/github';
+import { BuildSelector } from '../build-selector';
+
+export default class Container extends Component {
+	constructor( props ) {
+		super( props );
+
+		// Default build options to choose from.
+		const buildOptions = [
+			{
+				label: __( 'Latest release', 'amp-qa-tester' ),
+				value: 'release',
+				origin: 'release',
+			},
+			{
+				label: __( 'Develop branch', 'amp-qa-tester' ),
+				value: 'develop',
+				origin: 'branch',
+			},
+		];
+
+		this.state = {
+			isLoading: true,
+			isSwitching: false,
+			isDevBuild: false,
+			error: null,
+			buildOption: { label: '', value: '' },
+			buildOptions,
+		};
+
+		this.addReleaseBranchOptions();
+		this.addPullRequestOptions();
+	}
+
+	componentWillUnmount() {
+		clearTimeout( this.errorTimeout );
+		clearTimeout( this.pageReloadTimeout );
+	}
+
+	/**
+	 * Append to the default build options a list of PRs that have build zips available for download.
+	 */
+	addReleaseBranchOptions() {
+		// Retrieve the PRs from GitHub and append them to the list of builds above.
+		getProtectedBranches().then( ( branches ) => {
+			const releaseBranchNames = branches
+				.map( ( branch ) => branch.name )
+				.filter( ( branchName ) =>
+					/[0-9]+\.[0-9]+/.test( branchName )
+				);
+
+			// We only need the release branches 1.5 and after, since only they would have built zips.
+			const releasesWithBuilds = releaseBranchNames
+				.sort()
+				.slice( releaseBranchNames.indexOf( '1.5' ) );
+
+			const branchOptions = releasesWithBuilds.map( ( branch ) => {
+				return {
+					label: `${ branch } release branch`,
+					value: branch,
+					origin: 'branch',
+				};
+			} );
+
+			this.setState( {
+				buildOptions: this.state.buildOptions.concat( branchOptions ),
+			} );
+		} );
+	}
```

Comment:
I think these methods could be more readable using `async/await` syntax.

```suggestion
	async addReleaseBranchOptions() {
		// Retrieve the PRs from GitHub and append them to the list of builds above.
		const branches = await getProtectedBranches();

		const releaseBranchNames = branches
			.map( ( branch ) => branch.name )
			.filter( ( branchName ) => /[0-9]+\.[0-9]+/.test( branchName ) );

		// We only need the release branches 1.5 and after, since only they would have built zips.
		const releasesWithBuilds = releaseBranchNames
			.sort()
			.slice( releaseBranchNames.indexOf( '1.5' ) );

		const branchOptions = releasesWithBuilds.map( ( branch ) => {
			return {
				label: `${ branch } release branch`,
				value: branch,
				origin: 'branch',
			};
		} );

		this.setState( {
			buildOptions: this.state.buildOptions.concat( branchOptions ),
		} );
	}
```